src/Plugin/views/access/GroupPermission.php

```diff
@@ -0,0 +1,234 @@
+      $container->get('og.access'),
+      $container->get('og.context'),
+      $container->get('og.group_type_manager')
+    );
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function access(AccountInterface $account) {
+    $group = $this->getGroup();
+    if ($group instanceof ContentEntityInterface) {
+      list(, $permission) = explode(':', $this->options['perm']);
+      return $this->ogAccess->userAccess($group, $permission, $account);
+    }
+    return AccessResult::neutral();
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function alterRouteDefinition(Route $route) {
+    list($entity_type_id, $permission) = explode(':', $this->options['perm']);
+    $route->setRequirement('_og_user_access_group', $permission);
+    $route->setOption('_og_entity_type_id', $entity_type_id);
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function summaryTitle() {
+    $permissions = OptGroup::flattenOptions($this->getGroupPermissionOptions());
+    if (isset($permissions[$this->options['perm']])) {
+      return $permissions[$this->options['perm']];
+    }
+    return $this->options['perm'];
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function buildOptionsForm(&$form, FormStateInterface $form_state) {
+    parent::buildOptionsForm($form, $form_state);
+
+    $form['perm'] = [
+      '#type' => 'select',
+      '#options' => $this->getGroupPermissionOptions(),
+      '#title' => $this->t('Permission'),
+      '#default_value' => $this->options['perm'],
+      '#required' => TRUE,
+      '#description' => $this->t('Only users with the selected permission flag in a group retrieved from context will be able to access this display.'),
+
+    ];
+
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function getCacheMaxAge() {
+    return Cache::PERMANENT;
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function getCacheContexts() {
+    return ['user.permissions'];
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function getCacheTags() {
+    return [];
+  }
+
+  /**
+   * Returns the group that is relevant in the current context.
+   *
+   * @return \Drupal\Core\Entity\ContentEntityInterface|null
+   *   The group, or NULL if no group is found.
+   */
+  protected function getGroup() {
+    $contexts = $this->ogContext->getRuntimeContexts(['og']);
+    if (!empty($contexts['og']) && $group = $contexts['og']->getContextValue()) {
+      if ($group instanceof ContentEntityInterface) {
+        return $group;
+      }
+    }
+  }
+
+  /**
+   * Returns the full set of group permission options per entity type.
+   *
+   * @return array
+   *   A multidimensional array of group permissions grouped by the entity type
+   *   label.
+   */
+  protected function getGroupPermissionOptions() {
+    $permissions = [];
+    foreach ($this->groupTypeManager->getAllGroupBundles() as $group_entity_type_id => $group_bundles) {
+      $entity_type_label = (string) $this->entityTypeManager->getDefinition($group_entity_type_id)->getLabel();
```

Comment:
Why does this need to be cast to a string? According to the documentation of `EntityTypeInterface::getLabel()` it should always return a string.

Reply:
It's returning a TranslatableMarkup object, we should probably update that in `EntityTypeInterface`

Reply:
Created an issue for this: https://www.drupal.org/node/2921307

Reply:
I added a patch for #2921307.

Reply:
https://www.drupal.org/node/2921307 has been fixed.